test/integration/shells_int_test.go

```diff
@@ -0,0 +1,96 @@
+package integration
+
+import (
+	"runtime"
+	"testing"
+
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/stretchr/testify/suite"
+)
+
+type ShellsIntegrationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *ShellsIntegrationTestSuite) TestShells() {
+	suite.OnlyRunForTags(tagsuite.Critical, tagsuite.Shell)
+
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	var shells []e2e.Shell
+	switch runtime.GOOS {
+	case "linux":
+		shells = []e2e.Shell{e2e.Bash, e2e.Fish}
+	case "darwin":
+		shells = []e2e.Shell{e2e.Bash, e2e.Zsh, e2e.Tcsh}
+	case "windows":
+		shells = []e2e.Shell{e2e.Cmd}
+	}
+
+	// Checkout the first instance. It doesn't matter which shell is used.
+	cp := ts.SpawnWithOpts(e2e.WithArgs("checkout", "ActiveState-CLI/small-python"))
+	cp.Expect("Checked out project")
+	cp.ExpectExitCode(0)
+
+	for _, shell := range shells {
+		// Run the checkout in a particular shell.
+		cp := ts.SpawnInShell(
+			shell,
+			e2e.WithArgs("checkout", "ActiveState-CLI/small-python", string(shell)),
+		)
```

Comment:
It would be nice to set this up in a subtest so that only specific shells can be tested locally, and so that the output messages are organized more thoroughly.
```
	suite.T().Run(fmt.Sprintf("using_%s", shell), func(t *testing.T) {
```
I'm fairly sure that the design I had suggested would've allowed for that layer of nesting to be avoided, in case it's something you want to avoid (it is generally recommended to avoid nesting, but testing seems to be much looser regarding certain practices like this).

Reply:
That's a fair point. The checkout command checks a project out to a shell-specific folder, and I was using that to identify where in the loop a test might have been failing, but an explicit marker is probably better.